analytics.go

```diff
@@ -0,0 +1,78 @@
+package flagsmith
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+)
+
+const AnalyticsTimerInMilli = 10 * 1000
+const AnalyticsEndpoint = "analytics/flags/"
+
+type analyticDataStore struct {
+	mu   sync.Mutex
+	data map[int]int
+}
+type AnalyticsProcessor struct {
+	client   *resty.Client
+	store    *analyticDataStore
+	endpoint string
+}
+
+func NewAnalyticsProcessor(ctx context.Context, client *resty.Client, baseURL string, timerInMilli *int) *AnalyticsProcessor {
+	data := make(map[int]int)
+	dataStore := analyticDataStore{data: data}
+	tickerInterval := AnalyticsTimerInMilli
+	if timerInMilli != nil {
+		tickerInterval = *timerInMilli
+	}
+	processor := AnalyticsProcessor{
+		client:   client,
+		store:    &dataStore,
+		endpoint: baseURL + AnalyticsEndpoint,
+	}
+	go processor.start(ctx, tickerInterval)
+	return &processor
+}
+
+func (a *AnalyticsProcessor) start(ctx context.Context, tickerInterval int) {
```

Comment:
How come this is `start` whereas we capitalise the first letter of the rest of the functions here? e.g. `Flush`

Reply:
Ah, ok, is that a Go convention or does it have any actual functional effect?

Reply:
No, it does have functional effects, it prevents from doing 
`
	processor := NewAnalyticsProcessor(...)
	processor.start(...) // Will throw error


`
outside the package.